engine/src/cubos/engine/input/input.cpp

```diff
@@ -0,0 +1,197 @@
+#include <cubos/core/data/debug_serializer.hpp>
+#include <cubos/core/log.hpp>
+
+#include <cubos/engine/input/input.hpp>
+
+using cubos::core::data::Debug;
+using cubos::core::io::KeyEvent;
+using cubos::core::io::ModifiersEvent;
+using namespace cubos::engine;
+
+void Input::clear()
+{
+    mBoundActions.clear();
+    mBoundAxes.clear();
+    mBindings.clear();
+    CUBOS_DEBUG("Input bindings cleared");
+}
+
+void Input::removeBoundPlayer(std::vector<BindingIndex>& boundKeys, int player)
+{
+    boundKeys.erase(
+        std::remove_if(boundKeys.begin(), boundKeys.end(), [player](const auto& idx) { return idx.player == player; }),
+        boundKeys.end());
+}
```

Comment:
What in tarnation is this - at least separate it into two lines and add a comment explaining what this does